indigo/indigo/src/main/scala/indigo/shared/input/Mouse.scala

```diff
@@ -91,17 +121,13 @@ object Mouse {
 
   @tailrec
   private def isLeftMouseDown(isDown: Boolean, events: List[MouseEvent]): Boolean =
-    events match {
+    events match
       case Nil =>
         isDown
-
-      case MouseEvent.MouseDown(_) :: xs =>
+      case MouseEvent.MouseDown(_, MouseButton.LeftMouseButton) :: xs =>
         isLeftMouseDown(true, xs)
-
-      case MouseEvent.MouseUp(_) :: xs =>
+      case MouseEvent.MouseUp(_, MouseButton.LeftMouseButton) :: xs =>
         isLeftMouseDown(false, xs)
-
       case _ :: xs =>
         isLeftMouseDown(isDown, xs)
-    }
```

Comment:
Is this whole function (which I've never liked) now made redundant by some other function? I've got a feeling it isn't? But:

1. If it is redundant, could we add a deprecation annotation telling the user which function to use instead?
2. If it isn't... do we need one for right click? ...and the others? This does get used all over the place unfortunately.

Also if it's staying, suggestions for a better name welcome!

Reply:
This is used to work out if the left mouse has been continuously held down, in case that isn't clear.

Reply:
Revamped this function into a more generic one, in a similar fashion to what's done for the `Keyboard`... I'll push it and let you review that new impl 👀